Java/sereal/src/main/java/com/booking/sereal/Decoder.java

```diff
@@ -273,8 +288,6 @@ private void uncompressZlib() throws SerealException {
       this.data = outputStream.toByteArray();
     } catch (DataFormatException | IOException e) {
       throw new SerealException(e);
-    } finally {
-      inflater.end();
```

Comment:
Oh interesting, was this closing the inflater too early?

Reply:
The issue here is that the inflater is being reused. Closing it here was literally breaking the code.

So, `inflater.end()` is a good practice, simply not when the instance is reused :(